cyclops-ctrl/internal/cluster/k8sclient/client.go

```diff
@@ -273,6 +317,19 @@ func (k *KubernetesClient) GetManifest(group, version, kind, name, namespace str
 	return string(data), nil
 }
 
+func (k *KubernetesClient) Restart(group, version, kind, name, namespace string) (any, error) {
+	switch {
+	case isDeployment(group, version, kind):
+		return nil, k.RestartDeployment(group, version, kind, name, namespace)
+	case isDaemonSet(group, version, kind):
+		return nil, k.RestartDaemonsets(group, version, kind, name, namespace)
+	case isStatefulSet(group, version, kind):
+		return nil, k.RestartStatefulSets(group, version, kind, name, namespace)
+	}
+
+	return nil, nil
```

Comment:
Can we throw an error if provided group, version and kind cannot be restarted

```suggestion
	return errors.New(fmt.Sprintf("cannot restart: %v/%v %v %v/%v", group, version, kind, namespace, name))
```

Reply:
added